examples/platform/silabs/efr32/rs911x/hal/rsi_hal_mcu_ioports.c

```diff
@@ -83,17 +93,30 @@ void rsi_hal_config_gpio(uint8_t gpio_number, uint8_t mode, uint8_t value)
  */
 void rsi_hal_set_gpio(uint8_t gpio_number)
```

Comment:
can you refactor these functions to  not have 3 different versions of the same function with essentially the same switch case?

Reply:
modified the switch case to have a fall through
all the 3 api's have different interaction 
1. GPIO_PinModeSet
2. GPIO_PinInGet
3. GPIO_PinOutClear

Reply:
Mathieu comments is still valid. A single function that does all 3 would be cleaner.

Reply:
I am still not getting what do you mean to have a single function
`rsi_hal_config_gpio` `rsi_hal_set_gpio` `rsi_hal_get_gpio` these are called from the wiseconnect sdk. And the switch case is same but all 3 have different GPIO interactions i.e., 
`GPIO_PinModeSet` `GPIO_PinInGet` `GPIO_PinOutClear`
Can you provide a sample example about what you mean to modify

Reply:
ok, leave it as is. Just refactor the switch case with the suggestion i had left on another comment.